acestep/device_map/discovery.py

```
@@ -0,0 +1,60 @@
+    import torch
+
+    if not torch.cuda.is_available():
+        return []
+
+    gpus: List[GpuInfo] = []
+    for index in range(torch.cuda.device_count()):
+        props = torch.cuda.get_device_properties(index)
+        free_bytes, total_bytes = torch.cuda.mem_get_info(index)
+        try:
+            capability = torch.cuda.get_device_capability(index)
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)